compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -300,6 +300,7 @@ define_tables! {
     super_predicates: Table<DefIndex, Lazy!(ty::GenericPredicates<'tcx>)>,
     // As an optimization, a missing entry indicates an empty `&[]`.
     explicit_item_bounds: Table<DefIndex, Lazy!([(ty::Predicate<'tcx>, Span)])>,
+    is_trivial_mir: Table<DefIndex, Lazy<bool>>,
```

Comment:
Why do you use a `Lazy<bool>` here? Using `bool` directly would save space and be faster I think. Or maybe even `()` where a missing entry indicates false and an existing entry indicates true.

Reply:
I am using `Lazy<bool>` here because I don't know how any of this works :sweat_smile: using `Table<DefIndex, ()>` seems interesting. Will look into this